src/plugins/dashboard/public/dashboard_container/embeddable/integrations/controls/dashboard_control_group_integration.ts

```diff
@@ -65,12 +65,19 @@ export async function startControlGroupIntegration(
     return;
   }
 
-  this.untilInitialized().then(() => startSyncingDashboardControlGroup.bind(this)());
+  this.untilInitialized().then(() => {
+    const stopSyncingControlGroup =
+      startSyncingDashboardControlGroup.bind(this)()?.stopSyncingWithControlGroup;
+    this.onDestroyControlGroup = () => {
+      stopSyncingControlGroup?.();
+      this.controlGroup?.destroy();
+    };
```

Comment:
This fixes a bug where the control group wasn't actually getting destroyed, which meant it wasn't getting unmounted - this is part of the reason the popover stayed open when navigating away from the dashboard.